File: backend/app/schemas/careers.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


# Resume Schemas


class ResumeBase(BaseModel):
    filename: str
    original_name: str
    file_path: str
    file_size: int
    mime_type: str


class ResumeCreate(ResumeBase):
    user_id: str


class ResumeUpdate(BaseModel):
    parsed_skills: Optional[list[str]] = None
    parsed_experience: Optional[list[dict]] = None
    parsed_education: Optional[list[dict]] = None
    parsed_projects: Optional[list[dict]] = None
    parsed_summary: Optional[str] = None


class ResumeResponse(ResumeBase):
    model_config = ConfigDict(from_attributes=True)

    id: str
    user_id: str
    parsed_skills: list[str]
    parsed_experience: list[dict]
    parsed_education: list[dict]
    parsed_projects: list[dict]
    parsed_summary: Optional[str]
    created_at: datetime
    updated_at: datetime


# GmailConnection Schemas


class GmailConnectionBase(BaseModel):
    email: str


class GmailConnectionCreate(GmailConnectionBase):
    user_id: str


class GmailConnectionResponse(GmailConnectionBase):
    model_config = ConfigDict(from_attributes=True)

    id: str
    user_id: str
    scopes: list[str]
    is_active: bool
    emails_sent_today: int
    daily_limit: int
    token_expiry: Optional[datetime]
    created_at: datetime
    updated_at: datetime


# Recruiter Schemas


class RecruiterBase(BaseModel):
    name: str
    email: Optional[str] = None
    company: Optional[str] = None
    linkedin_url: Optional[str] = None


class RecruiterCreate(RecruiterBase):
    user_id: str


class RecruiterUpdate(BaseModel):
    is_verified: Optional[bool] = None
    notes: Optional[str] = None


class RecruiterResponse(RecruiterBase):
    model_config = ConfigDict(from_attributes=True)

    id: str
    user_id: str
    is_verified: bool
    notes: Optional[str]
    created_at: datetime
    updated_at: datetime


# RecruiterPost Schemas


class RecruiterPostBase(BaseModel):
    recruiter_id: str
    title: str
    company: str
    content: str
    post_url: Optional[str] = None


class RecruiterPostCreate(RecruiterPostBase):
    user_id: str


class RecruiterPostResponse(RecruiterPostBase):
    model_config = ConfigDict(from_attributes=True)

    id: str
    user_id: str
    location: Optional[str]
    required_skills: list[str]
    experience_level: Optional[str]
    salary_range: Optional[str]
    ats_match_score: Optional[int]
    is_saved: bool
    created_at: datetime


# Application Schemas


class ApplicationBase(BaseModel):
    recruiter_post_id: str
    resume_id: Optional[str] = None
    subject: Optional[str] = None
    body: Optional[str] = None


class ApplicationCreate(ApplicationBase):
    user_id: str


class ApplicationUpdate(BaseModel):
    status: Optional[str] = None
    subject: Optional[str] = None
    body: Optional[str] = None


class ApplicationResponse(ApplicationBase):
    model_config = ConfigDict(from_attributes=True)

    id: str
    user_id: str
    status: str
    ats_score: Optional[int]
    sent_at: Optional[datetime]
    replied_at: Optional[datetime]
    follow_up_sent: bool
    follow_up_at: Optional[datetime]
    created_at: datetime
    updated_at: datetime


# Analytics Schemas


class AnalyticsSummary(BaseModel):
    total_applications: int
    sent: int
    replied: int
    rejected: int
    no_response: int
    pending: int
    response_rate: float
    avg_ats_score: float

File: types/careers.ts
export interface Resume {
  id: string
  user_id: string
  filename: string
  original_name: string
  file_path: string
  file_size: number
  mime_type: string
  parsed_skills: string[]
  parsed_experience: Experience[]
  parsed_education: Education[]
  parsed_projects: Project[]
  parsed_summary: string | null
  created_at: string
  updated_at: string
}

export interface Experience {
  title: string
  company: string
  duration: string
  description?: string
}

export interface Education {
  degree: string
  institution: string
  year: string
}

export interface Project {
  name: string
  description?: string
  technologies?: string[]
}

export interface GmailConnection {
  id: string
  user_id: string
  email: string
  scopes: string[]
  is_active: boolean
  emails_sent_today: number
  daily_limit: number
  token_expiry: string | null
  created_at: string
  updated_at: string
}

export interface Recruiter {
  id: string
  user_id: string
  name: string
  email: string | null
  company: string | null
  linkedin_url: string | null
  is_verified: boolean
  notes: string | null
  created_at: string
  updated_at: string
}

export interface RecruiterPost {
  id: string
  user_id: string
  recruiter_id: string
  title: string
  company: string
  content: string
  post_url: string | null
  location: string | null
  required_skills: string[]
  experience_level: string | null
  salary_range: string | null
  ats_match_score: number | null
  is_saved: boolean
  created_at: string
}

export interface Application {
  id: string
  user_id: string
  recruiter_post_id: string
  resume_id: string | null
  subject: string | null
  body: string | null
  status: 'pending' | 'sent' | 'replied' | 'rejected' | 'no_response'
  ats_score: number | null
  sent_at: string | null
  replied_at: string | null
  follow_up_sent: boolean
  follow_up_at: string | null
  created_at: string
  updated_at: string
}

export interface AnalyticsSummary {
  total_applications: number
  sent: number
  replied: number
  rejected: number
  no_response: number
  pending: number
  response_rate: number
  avg_ats_score: number
}